### PR TITLE
DB: Bugfixes in `get.trait.data`

### PR DESCRIPTION
**Pull request by ashiklom**
Another bug found by @istfer. Basically, this code will fail if no traits have been found (common for, e.g., weird PFTs like SIPNET soil) because `trait.data` is an empty list (which breaks `dplyr::bind_rows`).

### CODE REVIEW

**Comment by ashiklom** [comment]
I found and fixed a few more bugs while testing this, so there a few more changes to review here.

File: base/db/R/get.trait.data.R [hunk 1]
```diff
@@ -189,26 +201,36 @@ get.trait.data.pft <- function(pft, modeltype, dbfiles, dbcon, trait.names,
...
-
-        if (nrow(diff_traits) > 0) {
-          diff_summary <- diff_traits %>%
-            dplyr::count(source, trait)
-          PEcAn.logger::logger.error(
-            "\n Prior has changed. \n",
-            "Here are the number of differing trait records by trait:\n",
-            PEcAn.logger::print2string(diff_summary),
-            wrap = FALSE
+        if (length(trait.data.check) != length(trait.data)) {
```
Comment (istfer) [inline base/db/R/get.trait.data.R]: I dont't think `trait.data` is defined yet

**Comment by istfer** [comment]
Also, the code thinks membership has changed, but it really is like a matter of reading it in

```
> diff_membership
    source   id   genus species scientificname
1 existing 1438 Unknown                Unknown
2  current 1438 Unknown    <NA>        Unknown
```

e.g if we change to `existing_membership <- utils::read.csv("/fs/data1/pecan.data/dbfiles/posterior/1000015654/species.csv",stringsAsFactors = FALSE, na.strings ="")` 

there is no difference anymore
```
> symmetric_setdiff(
+           dplyr::as_tibble(prior.distns, rownames = "trait"),
+           dplyr::as_tibble(existing_prior, rownames = "trait")
+         )
# A tibble: 0 x 6
# … with 6 variables: source <chr>, trait <chr>, distn <chr>, parama <dbl>, paramb <dbl>, n <int>
```

File: base/db/R/get.trait.data.R [hunk 2]
```diff
@@ -145,15 +152,20 @@ get.trait.data.pft <- function(pft, modeltype, dbfiles, dbcon, trait.names,
...
         # Check if PFT membership has changed
         PEcAn.logger::logger.debug("Checking if PFT membership has changed.")
-        existing_membership <- utils::read.csv(need_paths[["pft_membership"]],
-                                               stringsAsFactors = FALSE)
+        existing_membership <- utils::read.csv(
+          need_paths[["pft_membership"]],
+          # Columns are: id, genus, species, scientificname
+          # Need this so NA values are
+          colClasses = c("double", "character", "character", "character"),
+          stringsAsFactors = FALSE
```
Comment (istfer) [inline base/db/R/get.trait.data.R]:
```suggestion
          stringsAsFactors = FALSE,
          na.strings ="")
```

File: base/db/R/get.trait.data.R (hunk 1 above)
Comment (ashiklom) [reply base/db/R/get.trait.data.R]: Good catch. Fixing now.

**Comment by ashiklom** [comment]
@istfer OK, take 3. This one comes with a unit test, so I _think_ I've caught all the issues this time.

**Comment by ashiklom** [comment]
This currently passes all tests, but the [build breaks](https://travis-ci.org/PecanProject/pecan/jobs/546929342#L8428) because of an apparent change in `interactive.plotting.sda`. I ran `devtools::document` on this and it turned up no differences, so I think this must be a Travis bug (perhaps a mismatch in Roxygen versions?).

**Comment by tonygardella** [comment]
Looks like it's breaking with:
```
These files were changed by the build process:
 M docker/depends/pecan.depends
Have you run devtools::check and commited any updated Roxygen outputs?
The command "scripts/travis/script.sh" exited with 1.
```
Means you need to run `./scripts/generate_dependencies.R` and push up the changes to `pecan.depends`. Double check the order it puts packages in too, that mix up of order made me go crazy trying to find why my builds were breaking.

**Pull request merged**